### PR TITLE
feat: Events model migration + entities + mappers + sync

### PR DESCRIPTION
**Pull request by asdfryan**
Fixes: #529

[Describe your change here]

## Test Plan

Testing locally

## Deployment instructions

[Add any special deployment instructions here]

### CODE REVIEW

File: packages/sync-workflows/activities/import_records.ts [hunk 1]
```diff
@@ -97,6 +97,15 @@ export function createImportRecords(
         );
         break;
       }
+      case 'event': {
+        const readable = await client.listEvents(updatedAfter);
```
Comment (albertyfwu) [inline packages/sync-workflows/activities/import_records.ts:101]: shouldn't this be await eventService.upsertRemoteEvents?

File: packages/core/mappers/event.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    remoteCreatedAt,
+    remoteUpdatedAt,
+    remoteWasDeleted,
+  }: CrmEventExpanded,
+  expandedAssociations: string[] = []
+): Event => {
+  const expandAccount = expandedAssociations.includes('account');
+  const expandContact = expandedAssociations.includes('contact');
+  const expandOwner = expandedAssociations.includes('owner');
+  const expandLead = expandedAssociations.includes('');
```
Comment (albertyfwu) [inline packages/core/mappers/event.ts:35]: is this intentional? missing `lead`?

File: packages/core/remotes/crm/salesforce/mappers.ts [hunk 3]
```diff
@@ -33,6 +39,52 @@ export const fromSalesforceUserToRemoteUser = (record: Record<string, any>): Rem
   };
 };
 
+export const fromSalesforceEventToRemoteEvent = (record: Record<string, any>): RemoteEvent => {
+  // In SFDC, WhoId can refer to either a contact or a lead
+  const remoteContactId = (record.WhoId as string).startsWith(CONTACT_ID_PREFIX) ? record.WhoId : null;
```
Comment (albertyfwu) [inline packages/core/remotes/crm/salesforce/mappers.ts:44]: can `whoId` be  null?

File: packages/core/services/common_models/event_service.ts [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+        opportunityId: {
+          not: null,
+        },
+      },
+      data: {
+        opportunityId: null,
+      },
+    });
+
+    await this.prisma.$executeRawUnsafe(`
```
Comment (albertyfwu) [inline packages/core/services/common_models/event_service.ts:250]: we really need to do some de-duping/abstraction of code later

**Review by albertyfwu — APPROVED** [review]

File: packages/core/remotes/crm/salesforce/mappers.ts (hunk 3 above)
Comment (asdfryan) [reply packages/core/remotes/crm/salesforce/mappers.ts:44]: yes

Comment (asdfryan) [reply packages/core/remotes/crm/salesforce/mappers.ts:44]: actually I think I'm wrong -- let me fix this

**Pull request merged**